### PR TITLE
[RA-1307] Adding is volume supported implementation

### PR DESCRIPTION
**Pull request by ikorneev**

### CODE REVIEW

File: Samples/VShadowVolumeShadowCopy/cpp/shadow.cpp [hunk 1]
```diff
@@ -674,6 +674,27 @@ int CommandLineParser::MainRoutine(vector<wstring> arguments)
...
+
+
+            // Initialize the VSS client
+            m_vssClient.Initialize(VSS_CTX_ALL);
+
+            WCHAR wszVolumePathName[MAX_PATH];
+            BOOL supported = TRUE;
+            BOOL bWorked = ::GetVolumePathName(arguments[1].c_str(), wszVolumePathName, MAX_PATH);
+
+
```
Comment (akhvoshch) [inline Samples/VShadowVolumeShadowCopy/cpp/shadow.cpp]: Avoid more than one empty line.

File: Samples/VShadowVolumeShadowCopy/cpp/shadow.cpp [hunk 2]
```diff
@@ -674,6 +674,27 @@ int CommandLineParser::MainRoutine(vector<wstring> arguments)
...
+            HRESULT is_supported_result = m_vssClient.IsVolumeSupported(wszVolumePathName, &supported);
+            
+            ft.WriteLine(L"\nSupported check is done, bool: %i, BASEIMAGE: %i", supported, is_supported_result== VSS_E_NESTED_VOLUME_LIMIT);
```
Comment (akhvoshch) [inline Samples/VShadowVolumeShadowCopy/cpp/shadow.cpp]: What is BASEIMAGE and why should you print here?

File: Samples/VShadowVolumeShadowCopy/cpp/shadow.cpp [hunk 3]
```diff
@@ -674,6 +674,27 @@ int CommandLineParser::MainRoutine(vector<wstring> arguments)
...
+
+            // Initialize the VSS client
+            m_vssClient.Initialize(VSS_CTX_ALL);
+
+            WCHAR wszVolumePathName[MAX_PATH];
+            BOOL supported = TRUE;
+            BOOL bWorked = ::GetVolumePathName(arguments[1].c_str(), wszVolumePathName, MAX_PATH);
+
+
+            HRESULT is_supported_result = m_vssClient.IsVolumeSupported(wszVolumePathName, &supported);
```
Comment (akhvoshch) [inline Samples/VShadowVolumeShadowCopy/cpp/shadow.cpp:689]: Shouldn't you check `is_supported_result` and return non-zero code in case of error?

**Review by akhvoshch — CHANGES_REQUESTED** [review]

File: Samples/VShadowVolumeShadowCopy/cpp/shadow.cpp (hunk 3 above)
Comment (ikorneev) [reply Samples/VShadowVolumeShadowCopy/cpp/shadow.cpp:689]: Nope, i need to check if this function returns VSS_E_NESTED_VOLUME_LIMIT

File: Samples/VShadowVolumeShadowCopy/cpp/shadow.cpp (hunk 1 above)
Comment (ikorneev) [reply Samples/VShadowVolumeShadowCopy/cpp/shadow.cpp]: Fixed

File: Samples/VShadowVolumeShadowCopy/cpp/shadow.cpp [hunk 4]
```diff
@@ -674,6 +674,37 @@ int CommandLineParser::MainRoutine(vector<wstring> arguments)
...
+            BOOL bWorked = ::GetVolumePathName(arguments[1].c_str(), wszVolumePathName, MAX_PATH);
+
+            HRESULT is_supported_result = m_vssClient.IsVolumeSupported(wszVolumePathName, &supported);
+
+            switch (is_supported_result)
+            {
+            case S_OK:
+                ft.WriteLine(L"\nSupported check is done, bool: %i, NESTED_VOLUME_LIMIT: %i", supported, false);
+                return 0;
+                break;
```
Comment (IgorIonov) [inline Samples/VShadowVolumeShadowCopy/cpp/shadow.cpp]: I think break here is redundant

File: Samples/VShadowVolumeShadowCopy/cpp/shadow.cpp [hunk 5]
```diff
@@ -674,6 +674,37 @@ int CommandLineParser::MainRoutine(vector<wstring> arguments)
...
+            case VSS_E_NESTED_VOLUME_LIMIT:
+                ft.WriteLine(L"\nSupported check is done, bool: %i, NESTED_VOLUME_LIMIT: %i", supported, true);
+                return 1;
+                break;
```
Comment (IgorIonov) [inline Samples/VShadowVolumeShadowCopy/cpp/shadow.cpp]: And here

File: Samples/VShadowVolumeShadowCopy/cpp/shadow.cpp [hunk 6]
```diff
@@ -674,6 +674,37 @@ int CommandLineParser::MainRoutine(vector<wstring> arguments)
...
+            WCHAR wszVolumePathName[MAX_PATH];
+            BOOL supported = TRUE;
+            BOOL bWorked = ::GetVolumePathName(arguments[1].c_str(), wszVolumePathName, MAX_PATH);
+
+            HRESULT is_supported_result = m_vssClient.IsVolumeSupported(wszVolumePathName, &supported);
+
+            switch (is_supported_result)
+            {
+            case S_OK:
+                ft.WriteLine(L"\nSupported check is done, bool: %i, NESTED_VOLUME_LIMIT: %i", supported, false);
```
Comment (IgorIonov) [inline Samples/VShadowVolumeShadowCopy/cpp/shadow.cpp]: Maybe create a separate value with
L"\nSupported check is done, bool: %i, NESTED_VOLUME_LIMIT: %i"

File: Samples/VShadowVolumeShadowCopy/cpp/shadow.cpp (hunk 4 above)
Comment (ikorneev) [reply Samples/VShadowVolumeShadowCopy/cpp/shadow.cpp]: +

File: Samples/VShadowVolumeShadowCopy/cpp/shadow.cpp [hunk 7]
```diff
@@ -674,6 +673,37 @@ int CommandLineParser::MainRoutine(vector<wstring> arguments)
...
+            // Initialize the VSS client
+            m_vssClient.Initialize(VSS_CTX_ALL);
+
+            WCHAR wszVolumePathName[MAX_PATH];
+            BOOL supported = TRUE;
+            BOOL bWorked = ::GetVolumePathName(arguments[1].c_str(), wszVolumePathName, MAX_PATH);
+
+            HRESULT is_supported_result = m_vssClient.IsVolumeSupported(wszVolumePathName, &supported);
+
+            wstring result_phrase = L"\nSupported check is done, bool: %i, NESTED_VOLUME_LIMIT: %i";
```
Comment (IgorIonov) [inline Samples/VShadowVolumeShadowCopy/cpp/shadow.cpp]: const?

**Review by IgorIonov — APPROVED** [review]

**Review by akhvoshch — APPROVED** [review]

**Pull request merged**